### PR TITLE
convert the build modules to the new diagnostics apis

### PR DESCRIPTION
**Pull request by tomerd**
motivation: adapt new diagnostics apis

changes:
* refactor the build  modules and relevant call sites to use new diagnostics api
* clean up assumptions about local file system as a default argument
* adjust call-sites and test

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

**Comment by tomerd** [comment]
rebased on https://github.com/apple/swift-package-manager/pull/3777

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@abertelrud @neonichu this is rebased on #3777 and ready to go

File: Sources/PackageLoading/Target+PkgConfig.swift [hunk 1]
```diff
@@ -252,17 +253,26 @@ public func removeDefaultFlags(cFlags: [String], libs: [String]) throws -> ([Str
...
+        metadata.targetName = targetName
+        return metadata
+    }
+}
 
-    public init(pcFile: String, target: String) {
-        self.pcFile = pcFile
-        self.target = target
+extension ObservabilityMetadata {
+    public var pcFile: String? {
```
Comment (abertelrud) [inline Sources/PackageLoading/Target+PkgConfig.swift:266]: Nice.

**Review by abertelrud — APPROVED** [review]

File: Sources/PackageLoading/Target+PkgConfig.swift (hunk 1 above)
Comment (tomerd) [reply Sources/PackageLoading/Target+PkgConfig.swift:266]: with the current architecture the clients would need to know what to look for in the metadata, and there is no tie in back to the scope. these extensions however are usually internal such that only if you import the right module you can access them. happy to do something better if you have an idea in mind.

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**